core/src/test/java/org/elasticsearch/index/query/InnerHitBuilderTests.java

```diff
@@ -312,7 +313,7 @@ public static InnerHitBuilder randomInnerHits(boolean recursive, boolean include
         innerHits.setScriptFields(new HashSet<>(scriptFields.values()));
         FetchSourceContext randomFetchSourceContext;
         if (randomBoolean()) {
-            randomFetchSourceContext = new FetchSourceContext(randomBoolean());
+            randomFetchSourceContext = new FetchSourceContext(true, Strings.EMPTY_ARRAY, Strings.EMPTY_ARRAY);
```

Comment:
Can you add more randomization here. You can build a random list of field names for the includes and the excludes for instance.